File: Problem Set 1/credit/credit.c
#include <cs50.h>
#include <stdio.h>

void check_valid(long credit_card);
int extract_numbers(long number);


int main(void)
{
    long credit_card  = get_long("Number: ");
    check_valid(credit_card);
}

void check_valid(long credit_card)
{
    long for_length = credit_card;
    long card_check = credit_card;
    int times_2 = 0;
    int non_times = 0;
    int i = 0;
    int total = 0;
    int multi = 0;
    int j, remainder, remainder_multi, times_2_last, times_2_last_number;

    // get the length of the credit card
    while(for_length > 0)
    {
        for_length = for_length / 10;
        i++;
    }

     // check to see if the card is invalid
    if (i != 13 && i != 15 && i != 16)
    {
        printf("INVALID\n");
        return;
    }
    do
    {
            // get the last remainder of last number in the credit card
            remainder = credit_card % 10;

            // store each non_multiply number into the array
            non_times += remainder;

            // go to next number in credit card
            credit_card /= 10;

            // get the next remainder of next number in the credit card
            remainder_multi = credit_card % 10;
            times_2 = remainder_multi;

            credit_card /= 10;

            // times these number by 2
            times_2 = times_2 * 2;

            // get the individual number of the numbers that after they have been multiplied by 2
            times_2_last = times_2 % 10;
            times_2_last_number = times_2 / 10;
            multi = multi + times_2_last + times_2_last_number;
    }
    while (credit_card != 0);

    total = multi + non_times;
    // printf("%d \n", total);

    // Check if the total can be divided by 10 evenly
    if (total % 10 == 0)
    {
        // get the first two numbers of the credit card
        do
        {
            card_check /= 10;
        }
        while (card_check > 100);

        // check if the card is a Visa
        if (card_check / 10 == 4)
        {
            printf("VISA\n");
        }

        // check if the card is a Amex
        else if (card_check / 10 == 3 && (card_check % 10 == 4 || card_check % 10 == 7))
        {
            printf("AMEX\n");
        }
        // Check if the card is a Mastercard
        else if (card_check / 10 == 5 && (card_check % 10 > 0 && card_check % 10 < 6))
        {
            printf("MASTERCARD\n");
        }
        else
        {
            printf("INVALID\n");
        }
    }
    // if nothing matches then it must be an invalid card
    else
    {
        printf("INVALID\n");
    }
}

File: Problem Set 6/sentimental-credit/credit.py
# TODO
from cs50 import get_int
from cs50 import get_string
import re

non_multi = 0
multi = 0

credit_card = get_int("Number: ")

# to check the
check = credit_card

length = 0

# get the length of the card
for i in range(len(str(credit_card))):
    length += 1

# chcek if the lenght of the credit card is valid
if (length != 13 and length != 15 and length != 16):
    print("INVALID")

# while loop to look at each number
while True:
    # if the credit card number is 0 break out of the loop
    if credit_card == 0:
        break
    # add the first and then every other number to non mutltiply int
    non_multi += int(credit_card % 10)

    # divide the credit card by 10 to look at the next number
    credit_card = int(credit_card/10)

    # add the second and then every other number to times 2 int
    times2 = int((credit_card % 10) * 2)

    # divide the credit card by 10 to look at the next number
    credit_card = int(credit_card/10)

    # get the individual number of the numbers that after they have been multiplied by 2
    tmp = int(times2 % 10)
    tmp1 = int(times2 / 10)
    multi = int(multi) + tmp + tmp1

# if mutli and non_multi added up with remainder 10 = 0 print which card the credit card number belongs to
if (multi + non_multi) % 10 == 0:
    if int(str(check)[:1]) == 4:
        print("VISA")
    elif int(str(check)[:1]) == 5 and (int(str(check)[1:2]) >= 0 and int(str(check)[1:2]) <= 5):
        print("MASTERCARD")
    elif int(str(check)[:1]) == 3 and (int(str(check)[1:2]) == 4 or int(str(check)[1:2]) == 7):
        print("AMEX")
    else:
        print("INVALID")
else:
    print("INVALID")